crates/shim/src/mount.rs

```diff
@@ -552,6 +560,11 @@ pub fn mount_rootfs(
         None
     };
 
+    if let Some(workdir) = chdir {
+        env::set_current_dir(Path::new(&workdir)).unwrap_or_else(|_| {
```

Comment:
Changing the working directory of the current thread without changing back may pollute it. Should fork new process and change its working directory, otherwise, considering [Unshare(CLONE_FS)](https://github.com/containerd/containerd/pull/7513)

Reply:
This function is called in tokio::task::spawn_blocking, spawn_blocking is a separate thread, not a worker in tokio, and tokio does not schedule other asynchronous functions to this thread when it is asynchronously scheduled, and which other asynchronous functions are only scheduled in the worker thread. The spawn_blocking thread is destroyed by Tokio every few seconds after the execution is complete. Is it possible to reduce the extra unnecessary overhead by not creating a new thread here? @Burning1020

Reply:
Yes, I understand this function is running in another thread, but changing workding directory of one thread will affect another threads, because all threads in the same process having the same working directory. Here is an example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=82ec578917c982ee95e37d5cd3a6ac26

Reply:
Thank you for your suggestion, I will change this.